### ISSUE TITLE
Explicitly turn off authentication response caching

### ISSUE THREAD

**Issue by ViliusRuskys** [aspnet/AspNetCore#8739]
### Is your feature request related to a problem? Please describe.
We have a custom made batch-request middleware and need the ability to authenticate every sub request separately. Currently the aspnet authentication middleware authenticates the request only once per scope and all the subsequent calls return the same authentication result, which causes all of the sub-requests to have the same authentication result as the first sub-request.

### Describe the solution you'd like
Currently the authentication result caching happens in `AuthenticationHandler`'s `HandleAuthenticateOnceAsync()` method. It would be nice to have an exposed method to disable this behavior explicitly, so that authentication could be handled on all sub-requests separately.

**Comment by JunTaoLuo** [comment]
Sounds like an ordering issue. The auth middleware should be after the batch middleware and the batch middleware should ensure an appropriate scope is available to each sub-request.

**Comment by SlowLogicBoy** [comment]
We have registered, authentication middleware after Batch middleware, the problem is:
[HandleAuthenticateOnceAsync](https://github.com/aspnet/AspNetCore/blob/master/src/Security/Authentication/Core/src/AuthenticationHandler.cs#L164) method, it caches `Task<AuthenticationResult>` after first request from batch, and all subsequent requests use the same `AuthenticationResult`, where we need to authenticate separately every request from batch.

**Comment by Eilon** [comment]
Hi, batching scenarios are very complex and there is no direct support for them in ASP.NET Core. To properly support this you'll likely need to have custom DI scopes to make sure each sub-request gets its own view of the services, and you'll have to manage those lifetimes.

Unfortunately, this is outside of the scope of what we are able to support in this forum. While we do our best to look through all the issues filed here, to get a faster response we suggest posting your questions to [StackOverflow](https://www.stackoverflow.com) using the `asp.net-core` tag or contacting [Microsoft Support](https://support.microsoft.com/) for additional assistance.
